syntax = "proto3";

package mykafka;

message Error
{
  enum ErrCode
  {
    OK = 0;
    INVALID_FILENAME = 1;
    FILE_NOT_FOUND = 2;
    FILE_ERROR = 3;
    INDEX_ERROR = 4;
    LOG_ERROR = 5;
    PARTITION_ERROR = 6;
    NOT_FOUND = 7;
  }
  ErrCode code = 1;
  string msg = 2;
}

message SendMessageRequest
{
  int32 producer_id = 1;
  string topic = 2;
  int32 partition = 3;
  bytes payload = 4;
}

message SendMessageResponse
{
  Error error = 1;
  int64 offset = 2;
}

message GetMessageRequest
{
  int32 consumer_id = 1;
  string topic = 2;
  int32 partition = 3;
  int64 offset = 4;
}

message GetMessageResponse
{
  Error error = 1;
  bytes payload = 2;
}

service Broker
{
  rpc SendMessage(SendMessageRequest) returns (SendMessageResponse) {}
  rpc GetMessage(GetMessageRequest) returns (GetMessageResponse) {}
}
